kernel/kernel-api/src/main/java/io/delta/kernel/internal/util/InCommitTimestampUtils.java

```diff
@@ -0,0 +1,82 @@
+/*
+ * Copyright (2023) The Delta Lake Project Authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.delta.kernel.internal.util;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Optional;
+
+import io.delta.kernel.engine.Engine;
+import io.delta.kernel.internal.SnapshotImpl;
+import io.delta.kernel.internal.TableConfig;
+import io.delta.kernel.internal.actions.Metadata;
+import static io.delta.kernel.internal.TableConfig.IN_COMMIT_TIMESTAMPS_ENABLED;
+
+public class InCommitTimestampUtils {
+
+    /** Returns true if the current transaction implicitly/explicitly enables ICT. */
+    public static boolean didCurrentTransactionEnableICT(
+            Engine engine,
+            Metadata currentTransactionMetadata,
+            SnapshotImpl readSnapshot) {
+        // If ICT is currently enabled, and the read snapshot did not have ICT enabled,
+        // then the current transaction must have enabled it.
+        // In case of a conflict, any winning transaction that enabled it after
+        // our read snapshot would have caused a metadata conflict abort
+        // (see [[ConflictChecker.checkNoMetadataUpdates]]), so we know that
```

Comment:
ConflictChecker.checkNoMetadataUpdates does not seem to exist. Did you mean `ConflictChecker.handleMetadata`?

Reply:
Yes. Thanks for pointing this out.